### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by pmacro**
Some of the dependencies we're using are too old to be resolvable, which is breaking the build.  Some of the changes in this PR have been made (Xcode 10.0 -> 10.3) in order to bring the CircleCI environments for the supported Xcode versions closer together, which makes maintenance a little easier.

This PR:

- Updates Cocoapods and Ruby versions to match what we use elsewhere within Scoop.
- Drops Xcode 9 and adds Xcode 11 support.
- Tweaks simulator versions in order to establish a working, reliable test configuration.
- Update to Swift 5 (only 1 source change made, which is to test code).

### CODE REVIEW

File: SwiftyRSA.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -444,6 +444,7 @@
 			developmentRegion = English;
 			hasScannedForEncodings = 0;
 			knownRegions = (
+				English,
```
Comment (nickm01) [inline SwiftyRSA.xcodeproj/project.pbxproj:447]: @pmacro where did this change come from?

Comment (pmacro) [reply SwiftyRSA.xcodeproj/project.pbxproj:447]: Touching the Xcode project at all adds this, so probably shouldn't fight against that.  I tested removing this, then making some insignificant Xcode project change, and it reappears.

**Comment by pmacro** [comment]
> @pmacro If we are dropping support for Swift 3/Xcode 9 we need to update the `README.md`.
> Any other ways we should be communicating this change to the community?

Thanks for catching that.  I think the readme should be enough, as the old version will still be accessible via Cocoapods, Carthage, and the repo is well organized, with release tags etc.  So hopefully no-one will be left behind by this change.

File: README.md [hunk 2]
```diff
@@ -16,7 +16,7 @@ SwiftyRSA is used in the [Scoop](https://www.takescoop.com/) [iOS app](https://i
 Installation
 ------------
 
-### Swift 3.2 / 4.0+
+### Swift 5.0+
```
Comment (nickm01) [inline README.md:19]: @pmacro so what do you do if you want to use Swift 4.0+ or xcode 9?  Not clear whats supported

Comment (pmacro) [reply README.md:19]: For the Swift version, we are stating here what's supported.  So anything <5.0 is unsupported.  We'll tag a release with Swift 5 support, which should make it easy to see when we moved to Swift 5, but would rather not publicize old builds/branches etc. that support other versions of Swift when we're not prepared to provide support.  Might be time to remove the existing section on `2.3`, but that's not really part of this PR.

We don't mention the Xcode version at all today, so I've stuck with communicating support in Swift-version terms only.

Comment (pmacro) [reply README.md:19]: Made some changes here per offline discussion.

**Pull request merged**